src/web-auth/index.js

```diff
@@ -160,6 +160,31 @@ WebAuth.prototype.parseHash = function(options, cb) {
   ) {
     return cb(null, null);
   }
+  var responseType = (this.baseOptions.responseType || options.responseType || '').split(' ');
```

Comment:
because responseType is not a required field for `WebAuth` (since you can add responseType to other methods individually) and was also never required for `parseHash` as well, I'm making this check optional. It will only run the check when you set a responseType in either the `WebAuth` constructor or `parseHash` method.

Reply:
There is no default?

Reply:
not in auth0.js

Reply:
`response_type` is a required field anyway on the API, so even if "other methods can set it", shouldn't you check that it was set? 
source ->https://auth0.com/docs/api/authentication#social

Reply:
we have this in our docs: 
![image](https://user-images.githubusercontent.com/941075/37487940-e81c55c2-2871-11e8-8563-7730d519d409.png)


Because of this, people might not be adding responseType to the constructor call, only to the auth request calls.
Since `responseType` was never an option in the `parseHash` method,  we have to deal with the fact that maybe responseType is not defined.

Reply:
Let's go back.
- API Docs say it's required when you call the endpoints.
- Auth0JS docs say it's required as well, since "if you don't specify it here you must specify it on each method".

In which case will Auth0JS work "without a response type being set anywhere" and parseHash be called to obtain a "valid output"?

Reply:
the scenario we're handling is:

```js
//responseType is not required here
var auth0 = new auth0.WebAuth({
  domain: "{YOUR_AUTH0_DOMAIN}",
  clientID: "{YOUR_AUTH0_CLIENT_ID}"
});

//if you don't use a responseType here, we throw an error
auth0.authorize({responseType: 'token'})

//inside this method, responseType will be undefined
auth0.parseHash(function(err, authResult) {
  console.log(err, authResult);
});
```

Reply:
Thanks! that's super clear. 
So we're adding this check ONLY when a `response_type` is set on the `auth0` object, correct? I'm OK.

Reply:
either the WebAuth constructor or the parseHash method.

Reply:
I added a new jsdoc entry in the parseHash method for responseType: https://github.com/auth0/auth0.js/pull/707/files#diff-2f11667c8f671044416e38e4ceb9d7c8R126